README.md

```diff
@@ -139,55 +139,25 @@ import utc from '@momentjs/moment/timezones/utc/'; // Note trailing slash
 // Error: folders cannot be imported (there is no index.* magic)
 ```
 
-### 2. Default Field
+### 2. Conditional Mapping
 
-The `default` package.json field can be thought of as the "new main" for Node.js.
+Conditional mapping is an extension of `"exports"` that allows defining different mappings between different environments, for example having a different module path between Node.js and the browser.
 
-Like exports, the target value can be a string or array, must start with './' and be an exact file path - no extension searching applies.
-
-When there is no "default" field, the traditional "main" and its associated lookup will be used, including the default `index.js` main.
-
-"default" can also be set to _false_ to indicate that there is no main entry point for the package.
-
-If any value for "default" set set, the package is immediately encapsulated as if `"exports": {}` were set.
-
-#### Example
-
-```js
-{
-  "main": "./index.js",
-  "default": "./main.js"
-}
-```
-
-will load `pkg/main.js` from both CommonJS and ESM importers `require('pkg')` or `import 'pkg'`.
-
-```js
-{
-  "default": false
-}
-```
-
-will give an error - "No main entry point found for pkg" when trying to `require('pkg')` or `import 'pkg'`.
-
-### 3. Conditional Mapping
-
-Conditional mapping is an extension of the `"exports"` and `"default"` features that allows defining different mappings between different environments, for example having a different module path between Node.js and the browser.
-
-Conditional mappings are defined as _objects_ in the target slot for both `"exports"` and the `"default"` field.
+Conditional mappings are defined as _objects_ in the target slot for `"exports"`.
 
 The object has keys which are _condition names_, and values which correspond to the mapping target.
 
-Condition names are matched in a priority order. In Node.js the following conditions apply in priority order:
+In Node.js, condition names are matched in object order. The first condition name that matches a known condition name for the current environment will match. Node.js supports the following conditions:
```

Comment:
I think I'd want sorting the `package.json` (a common practice to reduce diff noise) to still work and not affect behavior. If the author wants to express a preferred order, they can always use the array syntax.

Reply:
I think there's a natural precedent order:

* `require`: Generally understood by node as a "subset of code in node.js". Most specific, match first.
* `node`: Any code meant for node.js specifically.
* `main`/`module`/`default`: Most generic, match last.

Reply:
I would expect `node` to match first if it's valid.  Prefer the ES module method when it's valid, `require` would be the fallback?

Reply:
To clarify why `require` has to be first: It will only be matched if the resolution is in the context of the CommonJS/require loader. That loader doesn't support ESM or other "modern" module formats. If it wouldn't match first, `require` would never be selected and those CommonJS contexts would always get a "broken" file they can't understand.

Reply:
^^^ Side note: This discussion is to be another reason not to let package authors figure out the correct priorities.

Reply:
`import from 'pkg'` does support CJS but it should prefer the ES module.  `require('pkg')` does not support ES modules so it should fail to match the modern formats even if higher priority.

Reply:
I think there's a misunderstanding about the `require` condition: It will never match for `import from 'pkg'` because it will only match loads via the `require` loader. And once we load and/or parse a file, we won't try others. We'll throw errors. This matching is happening 100% statically with local information, without trying to load resources or looking at file extensions etc..

> import from 'pkg' does support CJS but it should prefer the ES module.

To this sentence specifically: Import doesn't support *all* of CJS. For example it doesn't support arbitrary file extensions for CommonJS files and it doesn't support native modules (IIRC). So it's not just pragmatic that `import` ignores entries guarded by `require`, it's also arguably more correct. :)

Reply:
> I think there's a misunderstanding about the require condition: It will never match for import from 'pkg' because it will only match loads via the require loader.

Thanks, this makes sense and resolves my concern about your proposed priority order.

Reply:
Sure, I've added an update to go back to priority order here. Are you happy with fallbacks applying for conditions with invalid targets through the priority order? I quite like the idea of it but it could be worth checking it makes sense for use cases.